src/Common/src/Common.Security/ConfigurationExtensions.cs

```diff
@@ -10,63 +10,41 @@ namespace Steeltoe.Common.Security;
 public static class ConfigurationExtensions
 {
     /// <summary>
-    /// Adds the contents of pem encoded certificate and key files to configuration, for use with <see cref="CertificateOptions" />.
+    /// Adds file path information for a certificate and (optional) private key to configuration, for use with <see cref="CertificateOptions" />.
     /// </summary>
     /// <param name="builder">
     /// Your <see cref="IConfigurationBuilder" />.
     /// </param>
-    /// <param name="certFilePath">
-    /// The path on disk to locate a valid pem-encoded certificate file.
+    /// <param name="certificateName">
+    /// Name of the certificate.
     /// </param>
-    /// <param name="keyFilePath">
-    /// The path on disk to locate a valid pem-encoded RSA key.
+    /// <param name="certificateFilePath">
+    /// The path on disk to locate a valid certificate file.
     /// </param>
-    /// <param name="optional">
-    /// Whether or not to throw an exception if the files aren't found.
+    /// <param name="privateKeyFilePath">
+    /// The path on disk to locate a valid pem-encoded RSA key.
     /// </param>
-    public static IConfigurationBuilder AddPemFiles(this IConfigurationBuilder builder, string certFilePath, string keyFilePath, bool optional = false)
+    public static IConfigurationBuilder AddCertificate(this IConfigurationBuilder builder, string certificateName, string certificateFilePath,
```

Comment:
I'm wondering if we should make this internal. I get that it's convenient for CloudFoundry, as an alternative to implementing a `ConfigurationSource/Provider` that puts the file paths in `IConfiguration`, because the file paths can never change.

But I think we should tell people to put the paths in appsettings.json or Config Server, which is more flexible, instead of suggesting to use this method. What do you think?

Reply:
I hadn't given this enough thought; I think you're right that this does not need to be public and standard configuring rules should apply.